source/agora/api/handler/PreImageReceivedHandler.d

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+
+    Definitions of the PreImageReceivedHandler
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.api.handler.PreImageReceivedHandler;
+
+import vibe.web.rest;
+import vibe.http.common;
+
+public interface PreImageReceivedHandler
+{
+// The REST generator requires @safe methods
+@safe:
+
+    /***************************************************************************
+
+        Push a preImage in JSON format with HTTP POST
+
+        API:
+            POST /preimage_received
+
+    ***************************************************************************/
+
+    @method(HTTPMethod.POST)
+    @path("/preimage_received")
+    public void pushPreImage (string[string] preimage);
```

Comment:
Just make the type `Preimage` and let the json serializer do its work. There is no need to do it manually.